pkg/userdata/ubuntu/userdata.go

```diff
@@ -178,37 +178,188 @@ write_files:
   content: |
 {{ if ne .CloudConfig "" }}{{ .CloudConfig | indent 4 }}{{ end }}
 
+- path: "/etc/apt/sources.list.d/docker.list"
+  permissions: "0777"
+  content: deb [arch=amd64] https://download.docker.com/linux/ubuntu xenial stable
+
+- path: "/etc/apt/sources.list.d/crio.list"
+  permissions: "0777"
+  content: deb [arch=amd64] http://ppa.launchpad.net/projectatomic/ppa/ubuntu xenial stable
+
+- path: "/usr/local/bin/install-dependencies"
+  permissions: "0777"
+  content: |
+    #!/bin/bash
+    set -xeuo pipefail
+    apt-key add /opt/docker.asc
+    apt-key add /opt/crio.asc
+    apt update
+
+    {{- if .OSConfig.DistUpgradeOnBoot }}
+    apt dist-upgrade -y
+    {{- end }}
+    if [[ -x /var/run/reboot-required ]]; then
```

Comment:
That file is executable? I would expect it just exists or does not exist, so a [[ -e /var/run/reboot-required ]] would be appropriate, otherwise this will never match